llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -462,6 +462,16 @@ class TargetLoweringBase {
     return true;
   }
 
+  /// Return true if there is a sign extend on the input to this function. Used
+  /// to determine whether to transform the
+  /// @llvm.experimental.vector.partial.reduce.* intrinsic to
+  /// PARTIAL_REDUCE_SADD or PARTIAL_REDUCE_UADD. It also removes the extend
+  /// from the input. \p Input The 'Input' operand to the
+  /// @llvm.experimental.vector.partial.reduce.* intrinsic.
+  virtual bool isPartialReductionInputSigned(SDValue &Input) const {
+    return false;
+  }
```

Comment:
I just thought I'd do it this way in case other targets had a slightly different implementation of partial reductions? For example, originally I was also going to get it to remove the extends preceding dot products (before realising that I'd need more ISD nodes for sudot and usdot). But I thought I'd leave it like this in case other targets wanted to place other operations between the extend and the partial reduction?

Reply:
Also, in the future legalisation patch, I use it to see whether there is an `ISD::MUL` operation done on the input, which leads to different behaviour being done. This searches for the dot product IR pattern, which is target-specific.

Reply:
Removed this function, explained in `SelectionDAGBuilder.cpp` comment.